spec/instant.html

```diff
@@ -652,24 +652,21 @@ <h1>
         <dd>It computes the difference between the two times represented by _instant_ and _other_, optionally rounds it, and returns it as a Temporal.Duration object.</dd>
       </dl>
       <emu-alg>
+        1. If _operation_ is ~since~, let _sign_ be -1. Otherwise, let _sign_ be 1.
         1. Set _other_ to ? ToTemporalInstant(_other_).
-        1. If _operation_ is ~until~, then
-          1. Let _first_ be _instant_.
-          1. Let _second_ be _other_.
-        1. Else,
-          1. Let _first_ be _other_.
-          1. Let _second_ be _instant_.
         1. Set _options_ to ? GetOptionsObject(_options_).
         1. Let _smallestUnit_ be ? ToSmallestTemporalUnit(_options_, « *"year"*, *"month"*, *"week"*, *"day"* », *"nanosecond"*).
         1. Let _defaultLargestUnit_ be ! LargerOfTwoTemporalUnits(*"second"*, _smallestUnit_).
         1. Let _largestUnit_ be ? ToLargestTemporalUnit(_options_, « *"year"*, *"month"*, *"week"*, *"day"* », *"auto"*, _defaultLargestUnit_).
         1. Perform ? ValidateTemporalUnitRange(_largestUnit_, _smallestUnit_).
         1. Let _roundingMode_ be ? ToTemporalRoundingMode(_options_, *"trunc"*).
+        1. If _operation_ is ~since~, then
+          1. Set _roundingMode_ to ! NegateTemporalRoundingMode(_roundingMode_).
         1. Let _maximum_ be ! MaximumTemporalDurationRoundingIncrement(_smallestUnit_).
         1. Let _roundingIncrement_ be ? ToTemporalRoundingIncrement(_options_, _maximum_, *false*).
-        1. Let _roundedNs_ be ! DifferenceInstant(_first_.[[Nanoseconds]], _second_.[[Nanoseconds]], _roundingIncrement_, _smallestUnit_, _roundingMode_).
+        1. Let _roundedNs_ be ! DifferenceInstant(_instant_.[[Nanoseconds]], _other_.[[Nanoseconds]], _roundingIncrement_, _smallestUnit_, _roundingMode_).
         1. Assert: The following steps cannot fail due to overflow in the Number domain because abs(_roundedNs_) &le; 2 &times; nsMaxInstant.
-        1. Let _result_ be ! BalanceDuration(0, 0, 0, 0, 0, 0, _roundedNs_, _largestUnit_).
+        1. Let _result_ be ! BalanceDuration(0, 0, 0, 0, 0, 0, _sign_ &times; _roundedNs_, _largestUnit_).
         1. Return ! CreateTemporalDuration(0, 0, 0, 0, _result_.[[Hours]], _result_.[[Minutes]], _result_.[[Seconds]], _result_.[[Milliseconds]], _result_.[[Microseconds]], _result_.[[Nanoseconds]]).
```

Comment:
DifferenceTemporalPlainTime and DifferenceTemporalPlainDateTime perform BalanceDuration on the result and _then_ take _sign_ into account. Would that work here as well?
```suggestion
        1. Let _result_ be ! BalanceDuration(0, 0, 0, 0, 0, 0, _roundedNs_, _largestUnit_).
        1. Return ! CreateTemporalDuration(0, 0, 0, 0, _sign_ &times; _result_.[[Hours]], _sign_ &times; _result_.[[Minutes]], _sign_ &times; _result_.[[Seconds]], _sign_ &times; _result_.[[Milliseconds]], _sign_ &times; _result_.[[Microseconds]], _sign_ &times; _result_.[[Nanoseconds]]).
```
Or alternatively, to update those other operations to utilize _sign_ in BalanceDuration rather than CreateTemporalDuration?

Reply:
Yep, that would work, and would be even more consistent with other AOs.  But it would do 6 negations instead of only one. I assumed that highlighting this (small) optimization opportunity would be helpful for implementers, because it might not be obvious to readers that this opportunity exists given the complexity of these algorithms.

AFAIK, all the other `DifferenceTemporal*` AOs don't have this same optimization opportunity because their calls to BalanceDuration don't operate on a single number of nanoseconds.

But also I'd be OK with your suggestion because 6 negations vs. one negation seems like a trivially small optimization.

What do you think?  I'm happy with either option.

Reply:
I don't have a strong opinion either way so happy to follow @gibson042's suggestion. Latest commit includes that change.